### DISCUSSION TITLE
Issue with checkTree

### DISCUSSION THREAD

**Discussion by sreekanth-x-101** [rsuite/rsuite#3216] [Q&A]
The CheckTree works fine if I pass the data around 100 values. But the problem is we need to show approximately 10,000 values in the list.  it is failing to render when I am trying to render the CheckTree component with that huge data of around 10K values. I have tried virtualization as well. Is anyone facing the same issue?

**Comment by superman66** [comment]
Could you please give me more details regarding the rendering problem? Have you received any error messages, or could you provide me with a demo that can replicate the issue? @code-x-sreekanth

**Reply by sreekanth-x-101** [reply]
@superman66 
we are trying to render a checkTree like below:
[ ] CheckGroup 1
---[ ] Child A 
---[ ] Child B
---[ ] Child C
---[ ] list goes on based on the count of the child it could be any number but not more than a thousand children.
[ ] CheckGroup 2
---[ ] Child A 
---[ ] Child B
---[ ] Child C
---[ ] list goes on based on the count of the child it could be any number but not more than a thousand children.
[ ] CheckGroup 3
---[ ] Child A 
---[ ] Child B
---[ ] Child C
---[ ] list goes on based on the count of the child it could be any number but not more than a thousand children.

like above we are rendering about 10k checkboxes (assume 150 parent nodes with each having 80 children). while rendering the page the browser is not at all responding, it is taking a few seconds to render the checkTree, and on scrolling the checkTree the content of it is not rendering.
